client/src/hooks/useUnitHelper.ts

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { useUnitPreferences } from "../stores/PreferencesStore";
+import {
+  MeasurementTypes,
+  MeasurementTypesToConversionFunction,
+  MeasurementTypesToStorageUnit,
+} from "../utils/units";
+
+export default function useUnitHelper(type: MeasurementTypes) {
+  const unitPreferences = useUnitPreferences();
+
+  /**
+   * Given a value in storage units, convert it to display units based on the current user preferences.
+   */
+  const storageToDisplayUnit = React.useCallback(
+    (val: number): number => {
+      // TODO: figure out how to make this type-safe
+      const conversionFunc: any = MeasurementTypesToConversionFunction[type];
```

Comment:
Had to use `:any` here because I couldn't get typescript to understand that the storage unit and unit preferences were of the same type. Definitely open to suggestions here